examples/pytorch/rgcn/experimental/README.md

```diff
@@ -1,62 +1,64 @@
+This is an example of training RGCN node classification in a distributed fashion. Currently, the example only support training RGCN graphs with no input features. The current implementation follows [../rgcn/entity_claasify_mp.py](../rgcn/entity_claasify_mp.py).
 
-Before training, please install some python libs by pip:
+Before training, install python libs by pip:
 
 ```bash
-sudo pip3 install ogb
-sudo pip3 install pyinstrument
-sudo pip3 install pyarrow
+pip3 install ogb pyinstrument
 ```
 
-To use the example and tools provided by DGL, please clone the DGL Github repository.
-
-```bash
-git clone --recursive https://github.com/dmlc/dgl.git
-```
-Below, we assume the repository is cloned in the home directory.
-
 To train RGCN, it has four steps:
 
-### Step 0: set IP configuration file.
+### Step 0: Setup a Distributed File System 
+* You may skip this step if your cluster already has folder(s) synchronized across machines. 
 
-User need to set their own IP configuration file before training. For example, if we have four machines in current cluster, the IP configuration
-could like this:
+To perform distributed training, files and codes need to be accessed across multiple machines. A distributed file system would perfectly handle the job (i.e., NFS, Ceph). 
 
+Here is an example of how to setup NFS. First, install essential libs on the storage server
 ```bash
-172.31.0.1
-172.31.0.2
-172.31.0.3
-172.31.0.4
-```
+sudo apt-get install nfs-kernel-server
+``` 
 
-Users need to make sure that the master node (node-0) has right permission to ssh to all the other nodes.
+Then create a folder for sharing and (optional) remove all restrictions
+```bash
+sudo mkdir -p /NFS
+# Change ownership and remove restrictions
+sudo chown -R nobody:nogroup /NFS
+sudo chmod 755 /NFS
+```
 
-### Step 1: partition the graph.
+We assume that the all servers are under a subnet with ip range `192.168.0.0` to `192.168.255.255`. The exports configuration needs to be modifed to 
 
-The example provides a script to partition some builtin graphs such as ogbn-mag graph.
-If we want to train RGCN on 4 machines, we need to partition the graph into 4 parts.
+```bash
+sudo vim /etc/exports
+# add the following line 
+/NFS  192.168.0.0/16(rw,sync,no_subtree_check)
+```
 
-In this example, we partition the ogbn-mag graph into 4 parts with Metis. The partitions are balanced with respect to
-the number of nodes, the number of edges and the number of labelled nodes.
+The server's internal ip can be checked  via `ifconfig` or `ip`. If the ip does not begin with `192.168`, then you may use
 ```bash
-python3 partition_graph.py --dataset ogbn-mag --num_parts 4 --balance_train --balance_edges
+# for ip range 10.0.0.0 – 10.255.255.255	
+/NFS  10.0.0.0/8(rw,sync,no_subtree_check)
+# for ip range 172.16.0.0 – 172.31.255.255	
+/NFS  172.16.0.0/12(rw,sync,no_subtree_check)
```

Comment:
after modifying /etc/exports, do we need to restart the server?

we need to provide instructions of mounting NFS from the client.